File: java/counting_sort.java
import java.util.*;

public class counting_sort {

    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
        int n = scan.nextInt();
        char[] hike = scan.next().toCharArray();
        
        int count = 0;
        int altitude = 0;
        
        for(char c : hike) {
            // Step up
            if(c == 'U') {
                if(altitude == -1) {
                    count++;
                }
                altitude++;
            }
            // Step down
            else {
                altitude--;
            }
        }

        scan.close();
        
        System.out.println(count);
    }
}

File: swift/counting_sort.swift
var n = Int(readLine()!)!
var hike = readLine()!
var numValleys = 0
var altitude = 0

for char in hike.characters {
    if(char == "U") {
        if(altitude == -1) {
            numValleys += 1;
        }
        altitude += 1;
    }
    // Step down
    else {
        altitude -= 1;
    }
}

print(numValleys)